resources/codegen_inputs/colr.rs

```diff
@@ -0,0 +1,859 @@
+
+#![parse_module(read_fonts::tables::colr)]
+
+/// [COLR (Color)](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#colr-header) table
+table Colr {
+    /// Table version number - set to 0 or 1.
+    #[version]
+    version: BigEndian<u16>,
+    /// Number of BaseGlyph records; may be 0 in a version 1 table.
+    num_base_glyph_records: BigEndian<u16>,
+    /// Offset to baseGlyphRecords array (may be NULL).
+    #[nullable]
+    #[read_offset_with($num_base_glyph_records)]
+    base_glyph_records_offset: BigEndian<Offset32<BaseGlyphRecordArray>>,
+    /// Offset to layerRecords array (may be NULL).
+    #[nullable]
+    #[read_offset_with($num_layer_records)]
+    layer_records_offset: BigEndian<Offset32<LayerRecordArray>>,
+    /// Number of Layer records; may be 0 in a version 1 table.
+    num_layer_records: BigEndian<u16>,
+    /// Offset to BaseGlyphList table.
+    #[available(1)]
+    #[nullable]
+    base_glyph_list_offset: BigEndian<Offset32<BaseGlyphList>>,
+    /// Offset to LayerList table (may be NULL).
+    #[available(1)]
+    #[nullable]
+    layer_list_offset: BigEndian<Offset32<LayerList>>,
+    /// Offset to ClipList table (may be NULL).
+    #[available(1)]
+    #[nullable]
+    clip_list_offset: BigEndian<Offset32<ClipList>>,
+    /// Offset to DeltaSetIndexMap table (may be NULL).
+    #[available(1)]
+    #[nullable]
+    var_index_map_offset: BigEndian<Offset32>,
+    /// Offset to ItemVariationStore (may be NULL).
+    #[available(1)]
+    #[nullable]
+    item_variation_store_offset: BigEndian<Offset32>,
+}
+
+/// [BaseGlyphRecordArray](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyph-and-layer-records) table
+#[read_args(num_base_glyph_records: u16)]
+table BaseGlyphRecordArray {
```

Comment:
for these shim tables I might add a note in the docs explaining that it's a sort of a hack, and the rationale?

I also think in the medium term it probably makes sense to support 'offsets-to-non-tables' explicitly, but I'd have to think a bit about what this entails.

Reply:
The only place where it is dirty is in things like traversal, which is going to add an extra level of indirection? But I agree this case isn't too bad at all.